src/benchmarklib/table_generator.cpp

```diff
@@ -105,8 +107,30 @@ std::shared_ptr<Table> TableGenerator::generate_table(
 
   pseudorandom_engine.seed(rd());
 
+  // Base allocators if we don't use multiple NUMA nodes
+  auto allocator_ptr_base_column = PolymorphicAllocator<std::shared_ptr<BaseColumn>>{};
+  auto allocator_value_column_int = PolymorphicAllocator<ValueColumn<int>>{};
+  auto allocator_chunk = PolymorphicAllocator<Chunk>{};
+  auto allocator_int = PolymorphicAllocator<int>{};
+
   for (ChunkID chunk_index{0}; chunk_index < num_chunks; ++chunk_index) {
-    ChunkColumns columns;
+#if HYRISE_NUMA_SUPPORT
+    if (numa_distribute_chunks) {
```

Comment:
Is there any case in which we don't want the data to be distributed on a NUMA system?

I'm a bit unhappy with `numa_distribute_chunks` doing nothing if we don't have numa support. Failing is not a good option either.

Reply:
Hmm, I don't know. Since this is for benchmarking, I could see that you would want the option to have both, distributed, and not.

Reply:
Ok, but even then you should not be able to explicitly set an option that gets ignored in some cases. If `false` is the default, we could add a check that numa support has to be enabled for `true`.

Reply:
Not sure. The safe way is to throw an exception and wait for someone to complain about it.